x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/create_datasource_page/step_select_config.tsx

```diff
@@ -0,0 +1,253 @@
+
+  // Fetch agent configs info
+  const {
+    data: agentConfigsData,
+    error: agentConfigsError,
+    isLoading: isAgentConfigsLoading,
+  } = useGetAgentConfigs();
+  const agentConfigs = agentConfigsData?.items || [];
+  const agentConfigsById = agentConfigs.reduce(
+    (acc: { [key: string]: GetAgentConfigsResponseItem }, config) => {
+      acc[config.id] = config;
+      return acc;
+    },
+    {}
+  );
+
+  // Update parent package state
+  useEffect(() => {
+    if (packageInfoData && packageInfoData.response) {
+      updatePackageInfo(packageInfoData.response);
+    }
+  }, [packageInfoData, updatePackageInfo]);
+
+  // Update parent selected agent config state
+  useEffect(() => {
+    const fetchAgentConfigInfo = async () => {
+      if (selectedConfigId) {
+        setSelectedConfigLoading(true);
+        const { data, error } = await sendGetOneAgentConfig(selectedConfigId);
+        setSelectedConfigLoading(false);
+        if (error) {
+          setSelectedConfigError(error);
+          updateAgentConfig(undefined);
+        } else if (data && data.item) {
+          setSelectedConfigError(undefined);
+          updateAgentConfig(data.item);
+        }
+      } else {
+        setSelectedConfigError(undefined);
+        updateAgentConfig(undefined);
+      }
+    };
+    if (!agentConfig || selectedConfigId !== agentConfig.id) {
+      fetchAgentConfigInfo();
+    }
+  }, [selectedConfigId, agentConfig, updateAgentConfig]);
+
+  // Display package error if there is one
+  if (packageInfoError) {
+    return (
+      <Error
+        title={
+          <FormattedMessage
+            id="xpack.ingestManager.createDatasource.StepSelectConfig.errorLoadingPackageTitle"
+            defaultMessage="Error loading package information"
+          />
+        }
+        error={packageInfoError}
+      />
+    );
+  }
+
+  // Display agent configs list error if there is one
+  if (agentConfigsError) {
+    return (
+      <Error
+        title={
+          <FormattedMessage
+            id="xpack.ingestManager.createDatasource.StepSelectConfig.errorLoadingAgentConfigsTitle"
+            defaultMessage="Error loading agent configurations"
+          />
+        }
+        error={agentConfigsError}
+      />
+    );
+  }
+
+  return (
+    <EuiFlexGroup direction="column">
+      <EuiFlexItem>
+        <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <EuiTitle size="s">
+              <h3>
+                <FormattedMessage
+                  id="xpack.ingestManager.createDatasource.StepSelectConfig.selectAgentConfigTitle"
+                  defaultMessage="Select an agent configuration"
+                />
+              </h3>
+            </EuiTitle>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty iconType="plusInCircle" href={CREATE_NEW_CONFIG_URI} size="s">
```

Comment:
This should open new agent config flyout when #59479 is available

Reply:
I did not implement a route for the create flyout, but will now that I see your comment. Thinking route will be `.../configs?create`

Reply:
FYI - just [added](https://github.com/elastic/kibana/pull/59479/files#diff-901b60c97a27d2a24049b1ced7e4d301R179) support for this route. You should be able to use `useLink(AGENT_CONFIG_PATH + '?create')`

Reply:
Thanks Paul, but I believe the UX here is to embed the flyout component directly on this page

Reply:
oops. :) . I saw the `href` and though it was a link. Saw your comment on my PR to just keep it. 👍